### ISSUE TITLE
Do we fill Trilinos AMG constant modes directly from DoFHandler?

### ISSUE THREAD

**Issue by kronbichler** [dealii/dealii#7860]
In #7793, @jppelteret suggested to add a constructor of `TrilinosWrappers::PreconditionAMG::AdditionalData` that takes a `DoFHandlerType` and a `ComponentMask`, to simplify life for a user who might just want to use it. I was a bit sceptical because it pulls in a `DoFHandler` object into something that is otherwise pure linear algebra. Now we do not really strictly separate different parts, but I think this warrants discussion: We could either
* add the dof handler classes to `trilinos_precondition.h` (or the respective implementation file) and directly construct `AdditionalData` from `DoFHandler` as originally suggested,
* provide a convenience function to say `DoFTools` besides `extract_constant_modes` that immediately fills them into `PreconditionAMG::AdditionalData` (saves one line of code against the next option)
* or keep things basic without further simplification, i.e., a user first uses `DoFTools::extract_constant_modes` using the variable in `PreconditionAMG::AdditionalData`.

I think the current state is not overly complicated, but if there is a number of people who want to simplify this I'm perfectly fine with that as well (or @jppelteret did you want to fill in something else here)?

**Comment by tjhei** [comment]
My vote would be to keep things separate. I don't think the extra line of code is much of a problem.

**Comment by jppelteret** [comment]
Maybe we leave this issue open for one more day of discussion, but if it doesn't bother anyone then we can just leave everything as it is. I just found myself having forgotten to set the `constant_modes` for a (vector-valued) problem that otherwise matched all of the other default criterion. It took me a while to figure that out... It was a stupid mistake, but I reckoned that since we had all of the mechanics to automatically initialise this that we should.
